server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -143,6 +152,8 @@ public ExtensionsOrchestrator(Settings settings, Path extensionsPath) throws IOE
         this.clusterService = null;
         this.namedWriteableRegistry = null;
         this.listener = new ExtensionActionListener();
+        this.client = null;
+        this.extensionActions = null;
```

Comment:
Seems like spending time in a constructor setting lots of things to `null` (vs. just initializing it to null as a field and ignoring it here) isn't right.  If it's not `final` we don't need to handle it here, and handling it with `null` isn't really handling it.

Is this placeholder code for actually assigning something?  If we can't assign it now but it's assigned later, perhaps commenting `// will be initialized in fooMethod()` would be helpful.

Reply:
haha that is true. Mostly all of them are being initialized later. I can remove all of them which are null.

Reply:
Compiler only complains about unset `final` attributes.

Although in general, leaving things as `null` means you have to null-check it when used, thus the suggestion to add a comment to help others navigate the code